lib/ChunkGroup.js

```diff
@@ -0,0 +1,113 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Tobias Koppers @sokra
+*/
+"use strict";
+
+const SortableSet = require("./util/SortableSet");
+
+const getArray = set => Array.from(set);
+
+const sortById = (a, b) => {
+	if(a.id < b.id) return -1;
+	if(b.id < a.id) return 1;
+	return 0;
+};
+
+const getId = set => {
+	set.sort();
+	return Array.from(set, x => x.id).join("+");
+};
+
+const getDebugId = set => {
+	set.sort();
+	return Array.from(set, x => x.debugId).join("+");
+};
+
+class ChunkGroup {
+	constructor() {
+		this._chunks = new SortableSet(undefined, sortById);
+		this._parents = new SortableSet(undefined, sortById);
+	}
+
+	get debugId() {
+		return this._parents.getFromUnorderedCache(getDebugId);
+	}
+
+	get id() {
+		return this._parents.getFromUnorderedCache(getId);
```

Comment:
Why is the `id` derived from `_parents` instead of `_chunks`?

Reply:
Yeah, I did this wrong at first and used `_chunks` too.

The ChunkGroup groups all it's parents with logical and. So it has to be the parents.

Example:

```
[0] entry -> [2] on-demand
```

`->` means parent-child-connection

Now we split entry into vendor and entry

```
[0] entry -> ChunkGroup 1
[1] vendor -> ChunkGroup 1
ChunkGroup 1 -> [2] on-demand
```

The `id` for the ChunkGroup 1 would be the sum of it's parents `entry` and `vendor` => `[0+1]`

Note that these `id` is only used in the Stats output. It doesn't manifest itself in the bundle.js. ChunkGroups are not output related.